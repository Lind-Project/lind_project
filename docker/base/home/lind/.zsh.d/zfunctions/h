## Fix zsh annoying history behavior
 if [[ -z "$*" ]]; then history; else egrep "$*" <(fc -l 1); fi
