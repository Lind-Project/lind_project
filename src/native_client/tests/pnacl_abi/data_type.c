/*
 * Copyright (c) 2011 The Native Client Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file.
 *
 * tests whether our data type are lowered as expected (ILP32)
 * TODO: sizes of size_sb and size_ue are not quite right.
 */

#include <setjmp.h>
#include <stdarg.h>
#include <stddef.h>
#include <unwind.h>

int size_c  = sizeof(char);
int size_s  = sizeof(short);
int size_i  = sizeof(int);
int size_l  = sizeof(long);
int size_ll = sizeof(long long);

int size_p = sizeof(void*);

int size_f  = sizeof(float);
int size_d  = sizeof(double);
int size_ld = sizeof(long double);

/* A 16 byte buffer to accomodate all three architectures */
int size_vl = sizeof(va_list);

/* A 1024 byte buffer to accomodate future architectures */
int size_jb = sizeof(jmp_buf);

#define SET_ALIGNMENT(T, name) \
  typedef struct { char dummy; T x; } AlignStruct_ ## name; \
  int align_ ## name = offsetof(AlignStruct_ ## name, x)

SET_ALIGNMENT(char, c);
SET_ALIGNMENT(short, s);
SET_ALIGNMENT(int, i);
SET_ALIGNMENT(long, l);
SET_ALIGNMENT(long long, ll);

SET_ALIGNMENT(void*, p);

SET_ALIGNMENT(float, f);
SET_ALIGNMENT(double, d);
SET_ALIGNMENT(long double, ld);

SET_ALIGNMENT(va_list, vl);

SET_ALIGNMENT(jmp_buf, jb);
