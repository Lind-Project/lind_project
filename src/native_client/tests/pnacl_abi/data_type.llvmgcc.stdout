target triple = "armv7-none-linux-gnueabi"

@size_c = global i32 1
@size_s = global i32 2
@size_i = global i32 4
@size_l = global i32 4
@size_ll = global i32 8
@size_p = global i32 4
@size_f = global i32 4
@size_d = global i32 8
@size_ld = global i32 8
@size_vl = global i32 16
@size_jb = global i32 1024
@align_c = global i32 1
@align_s = global i32 2
@align_i = global i32 4
@align_l = global i32 4
@align_ll = global i32 8
@align_p = global i32 4
@align_f = global i32 4
@align_d = global i32 8
@align_ld = global i32 8
@align_vl = global i32 4
@align_jb = global i32 8
